b2b-sample/apps/business-admin-app/components/sections/sections/settingsSection/idpSection/otherComponents/idpCreateModal/externalIdentityProvider.tsx

```diff
@@ -0,0 +1,214 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IdentityProvider, IdentityProviderConfigureType, IdentityProviderTemplate } from
+    "@b2bsample/business-admin-app/data-access/data-access-common-models-util";
+import { controllerDecodeCreateIdentityProvider } from
+    "@b2bsample/business-admin-app/data-access/data-access-controller";
+import { FormButtonToolbar, FormField } from "@b2bsample/shared/ui/ui-basic-components";
+import { checkIfJSONisEmpty } from "@b2bsample/shared/util/util-common";
+import { LOADING_DISPLAY_BLOCK, LOADING_DISPLAY_NONE, fieldValidate } from
+    "@b2bsample/shared/util/util-front-end-util";
+import { Session } from "next-auth";
+import { useState } from "react";
+import { Form } from "react-final-form";
+import { Loader, Radio, RadioGroup } from "rsuite";
+import FormSuite from "rsuite/Form";
+import styles from "../../../../../../../styles/Settings.module.css";
+
+interface ExternalIdentityProviderProps {
+    session: Session
+    template: IdentityProviderTemplate
+    onIdpCreate: (response: IdentityProvider) => void,
+    onCancel: () => void,
+}
+
+/**
+* 
+* @param prop - `ExternalIdentityProviderProps`
+* 
+* @returns Form to create external idp
+*/
+export default function ExternalIdentityProvider(prop: ExternalIdentityProviderProps) {
+
+    const { session, template, onIdpCreate, onCancel } = prop;
+
+    const [ loadingDisplay, setLoadingDisplay ] = useState(LOADING_DISPLAY_NONE);
+    const [ configureType, setConfigureType ]
+        = useState<IdentityProviderConfigureType>(IdentityProviderConfigureType.AUTO);
+
+    const validate = (values: Record<string, string>): Record<string, string> => {
+        let errors: Record<string, string> = {};
+
+        errors = fieldValidate("application_name", values.application_name, errors);
+        errors = fieldValidate("client_id", values.client_id, errors);
+        errors = fieldValidate("client_secret", values.client_secret, errors);
+
+        switch (configureType) {
+            case IdentityProviderConfigureType.AUTO:
+                errors = fieldValidate("discovery_url", values.discovery_url, errors);
+
+                break;
+
+            case IdentityProviderConfigureType.MANUAL:
+                errors = fieldValidate("authorization_endpoint", values.authorization_endpoint, errors);
+                errors = fieldValidate("token_endpoint", values.token_endpoint, errors);
+
+                break;
+        }
+
+        return errors;
+    };
+
+    const onConfigureTypeChange = (value: IdentityProviderConfigureType): void => {
+        setConfigureType(value);
+    };
+
+    const onUpdate = async (values: Record<string, string>): Promise<void> => {
+        setLoadingDisplay(LOADING_DISPLAY_BLOCK);
+        controllerDecodeCreateIdentityProvider(session, template, values, configureType)
+            .then((response) => onIdpCreate(response))
```

Comment:
Why are we not handling errors here?

Reply:
The error is handled inside the controllerDecodeCreateIdentityProvider()